pkgs/applications/video/miru/default.nix

```diff
@@ -0,0 +1,41 @@
+{ appimageTools, lib, fetchurl }:
+let
+  pname = "miru";
+  version = "3.11.6";
+  sha256 = "0abc03c5a8cf9557309d054d2281b4267951f5d929c8ee11d4f015ed335ef2f2";
+  description = "Bittorrent streaming software for cats. Stream anime torrents, real-time with no waiting for downloads (If xdg-open isn't working on login, try using ungoogled-chromium)";
+  homepage = "https://github.com/ThaUnknown/miru/";
+  license = lib.licenses.gpl3Only;
+  filename = "linux-Miru-${version}.AppImage";
+in
+appimageTools.wrapType2 rec {
+  name = pname;
+  inherit pname;
+  inherit version;
+
+  src = fetchurl {
+    url = "${homepage}/releases/download/v${version}/${filename}";
+    inherit sha256;
+  };
+
+  extraInstallCommands =
+    let
+      contents = appimageTools.extractType2 { name = filename; inherit src; };
+    in
+    ''
+      mkdir -p "$out/share/applications"
+      mkdir -p "$out/share/lib/miru"
+      cp -r ${contents}/{locales,resources} "$out/share/lib/miru"
+      cp -r ${contents}/usr/* "$out"
+      cp "${contents}/${pname}.desktop" "$out/share/applications/"
+      substituteInPlace $out/share/applications/${pname}.desktop --replace 'Exec=AppRun' 'Exec=${pname}'
+    '';
+
+  meta = with lib; {
+    inherit description;
+    inherit homepage;
+    inherit license;
+    platforms = [ "x86_64-linux" ];
```

Comment:
Does it not work on macos or is it supposed to only work on linux?

Reply:
It's an AppImage, I don't know if those work in macos.

Reply:
> Does it not work on macos or is it supposed to only work on linux?

Appimages are linux-only. MacOS uses other format.

Reply:
then please change it to `platforms = platforms.linux;`

Reply:
I am not so sure, because there is more than x86_64 on (our?) Linux.

The elegant way is to intersect `linux` with `x86_64`.

Reply:
Ah I'm sorry it off-course only works on x86_64-linux.